src/hotspot/share/opto/vectorIntrinsics.cpp

```diff
@@ -432,7 +432,7 @@ bool LibraryCallKit::inline_vector_nary_operation(int n) {
       return false;
      }
   } else {
-    const TypeVect* vt = TypeVect::make(elem_bt, num_elem);
+    const TypeVect* vt = TypeVect::make(elem_bt, num_elem, is_vector_mask(vbox_klass));
```

Comment:
Are there any other nary mask operations besides the `"and/xor/or" `?

Reply:
andNot and eq, but they are implemented in terms of and/or/xor/not. So its more of a optimization since AVX512 have direct instructions for andNot operations on mask.

Reply:
So I think this change is not needed, since the vector mask type will be created in the `VectorNode::make` method, which is currently used for `and/or/xor`.